File: java/quantile-estimator/src/main/java/org/mydotey/quantile/value/DoubleCalculator.java
package org.mydotey.quantile.value;

/**
 * @author koqizhao
 *
 * Apr 1, 2018
 */
public class DoubleCalculator implements Calculator<Double> {

    public static final DoubleCalculator DEFAULT = new DoubleCalculator();

    @Override
    public Double add(Double t, Double t2) {
        return t + t2;
    }

    @Override
    public Double subtract(Double t, Double t2) {
        return t - t2;
    }

    @Override
    public Double multiply(Double t, double t2) {
        return t * t2;
    }

    @Override
    public Double divide(Double t, double t2) {
        return t / t2;
    }

}

File: dotnet/src/MyDotey.QuantileEstimator/Value/DoubleCalculator.cs
using System;

/**
 * @author koqizhao
 *
 * Apr 1, 2018
 */
namespace MyDotey.Quantile.Value
{
    public class DoubleCalculator : ICalculator<Double>
    {
        public static readonly DoubleCalculator DEFAULT = new DoubleCalculator();

        public virtual Double Add(Double t, Double t2)
        {
            return t + t2;
        }

        public virtual Double Subtract(Double t, Double t2)
        {
            return t - t2;
        }

        public virtual Double Multiply(Double t, double t2)
        {
            return t * t2;
        }

        public virtual Double Divide(Double t, double t2)
        {
            return t / t2;
        }
    }
}